org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                        
+                        string propertyOrAttribute = dvAttributes.GetValueOrNull("IsPropertyOrAttribute").Value;
+                        string hsKey = dvAttributes.GetValueOrNull("HubSpotAttributeKey").Value;
+                        string type = dvAttributes.GetValueOrNull( "Type" ).Value;
+                        string key = hsSyncDv.Value;
+                        var value = "";
+                        DateTime date;
+
+                        // Get person property or attribute
+                        if ( propertyOrAttribute == "Property" ) // is property
+                        {
+                            value = person.GetPropertyValue(key).ToStringSafe();
+                        }
+                        else // is attribute
+                        {
+                            var attributeQry = attributeService.Queryable().Where( a => a.EntityTypeId == 15 && a.Key == key ).AsNoTracking();
+                            try
+                            {
+                                Debug.WriteLine( "Type/Key: " + " " + type + " / " + hsKey );
+                                value = attributeValueService.Queryable().Where( av => av.EntityId == current_id ).Join( attributeQry, av => av.AttributeId, a => a.Id, ( av, a ) => av ).Select( av => av.Value ).AsNoTracking().FirstOrDefault().ToStringSafe();
+                            }
+                            catch
+                            {
+                                value = "";
+                            }
+                        }
+
+                        // Set date values to HubSpot required format
+                        if ( type == "Date" )
+                        {
+                            if ( string.IsNullOrEmpty(value) == false )
+                            {
+                                if ( value.AsDateTime() != null )
+                                {
+                                    value = ConvertDate( value.AsDateTime() );
+                                }
+                            }
+                        }
+
+                        // Patch it!
+                        Debug.WriteLine( "Patching: " + hsKey + " " + value );
+                        properties.Add( new HubspotPropertyUpdate() { property = hsKey, value = value } );
+
+                    }
+                    
+                    // Handle name changes
+                    properties.Add(new HubspotPropertyUpdate() { property = "firstname", value = person.NickName });
+                    properties.Add(new HubspotPropertyUpdate() { property = "lastname", value = person.LastName });
+
+                    // Handle Phone
+                    PhoneNumber mobile = person.PhoneNumbers.FirstOrDefault( n => n.NumberTypeValueId == 12 );
+                    if ( mobile != null && !mobile.IsUnlisted && mobile.IsMessagingEnabled )
```

Comment:
Please swap out the `!mobile.IsUnlisted` for `mobile.IsUnlisted == false` for readability.